src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java

```diff
@@ -201,32 +196,29 @@ private String getIteratorOutput(RuntimeIterator iterator, DynamicContext dynami
                 sb.append("\n");
                 itemCount++;
             }
+            if (iterator.hasNext() && itemCount == this.configuration.getResultSizeCap()) {
+                System.err.println(
+                    "Warning! The output sequence contains a large number of items but its materialization was capped at "
+                        + SparkSessionManager.COLLECT_ITEM_LIMIT
+                        + " items. This value can be configured with the --result-size parameter at startup"
+                );
+            }
```

Comment:
I believe showing this error after the query results would be much more noticeable to the users, but I'm not sure how much effort that would require

Reply:
I tried it but will require more efforts.